Find a functionally equivalent code:android.media.AudioService.setModeInt:COMMENT
<android.media.AudioService: int setModeInt(int,IBinder,int)>
Method Modifier: private     
Method Modifier: No          
Method Updated:  Yes         
Comment:// any mode other than NORMAL.

@@ -1,4 +1,7 @@
 {
+    if (DEBUG_MODE) {
+        Log.v(TAG, "setModeInt(mode=" + mode + ", pid=" + pid + ")");
+    }
     int newModeOwnerPid = 0;
     if (cb == null) {
         Log.e(TAG, "setModeInt() called with null binder");
@@ -24,29 +27,35 @@
                 hdlr = mSetModeDeathHandlers.get(0);
                 cb = hdlr.getBinder();
                 mode = hdlr.getMode();
+                if (DEBUG_MODE) {
+                    Log.w(TAG, " using mode=" + mode + " instead due to death hdlr at pid=" + hdlr.mPid);
+                }
             }
         } else {
             if (hdlr == null) {
@@ -44,49 +50,58 @@
         if (mode != mMode) {
             status = AudioSystem.setPhoneState(mode);
             if (status == AudioSystem.AUDIO_STATUS_OK) {
+                if (DEBUG_MODE) {
+                    Log.v(TAG, " mode successfully set to " + mode);
+                }
                 mMode = mode;
             } else {
                 if (hdlr != null) {
@@ -51,56 +60,68 @@
                     cb.unlinkToDeath(hdlr, 0);
                 }
                 // force reading new top of mSetModeDeathHandlers stack
+                if (DEBUG_MODE) {
+                    Log.w(TAG, " mode set to MODE_NORMAL after phoneState pb");
+                }
                 mode = AudioSystem.MODE_NORMAL;
             }
         } else {
@@ -66,75 +78,83 @@
             }
         }
         int streamType = getActiveStreamType(AudioManager.USE_DEFAULT_STREAM_TYPE);
-        if (streamType == STREAM_REMOTE_MUSIC) {
-            // here handle remote media playback the same way as local playback
-            streamType = AudioManager.STREAM_MUSIC;
-        }
         int device = getDeviceForStream(streamType);
         int index = mStreamStates[mStreamVolumeAlias[streamType]].getIndex(device);
         setStreamVolumeInt(mStreamVolumeAlias[streamType], index, device, true);

